src/Compilers/Core/Portable/Syntax/ChildSyntaxList.cs

```diff
@@ -83,17 +83,26 @@ private static int Occupancy(GreenNode green)
             return green.IsList ? green.SlotCount : 1;
         }
 
+        internal record struct SlotData(int SlotIndex, int FirstIndexAtSlotIndex, int PositionAtSlotIndex);
+
+        internal static SyntaxNodeOrToken ItemInternal(SyntaxNode node, int index)
+        {
+            SlotData slotData = new SlotData(SlotIndex: 0, FirstIndexAtSlotIndex: 0, PositionAtSlotIndex: node.Position);
+
+            return ItemInternal(node, index, ref slotData);
+        }
+
         /// <summary>
         /// internal indexer that does not verify index.
         /// Used when caller has already ensured that index is within bounds.
         /// </summary>
-        internal static SyntaxNodeOrToken ItemInternal(SyntaxNode node, int index)
+        internal static SyntaxNodeOrToken ItemInternal(SyntaxNode node, int index, ref SlotData slotData)
         {
             GreenNode? greenChild;
             var green = node.Green;
-            var idx = index;
-            var slotIndex = 0;
-            var position = node.Position;
+            var idx = index - slotData.FirstIndexAtSlotIndex;
```

Comment:
can you comment this?  not sure i understand it.

Reply:
Done